src/library_webgl.js

```diff
@@ -1767,7 +1767,7 @@ var LibraryGL = {
 
 #if MAX_WEBGL_VERSION >= 2
     if ({{{ isCurrentContextWebGL2() }}}) { // WebGL 2 provides new garbage-free entry points to call to WebGL. Use those always when possible.
-      if (data) {
+      if (data && size > 0) {
```

Comment:
Can you add at the top a

```js
#if ASSERTIONS
  assert(size >= 0);
#endif
```
and then here do just `if (data && size)`.

We do not currently support larger than 2GB buffers (that needs special handling, which I doubt anyone will use with WebGL - those use cases will be with Wasm64 and WebGPU)